Fix Stack link

bronzehedwick opened the pull request:
Without the `https://` in the beginning, Github turns urls into relative links inside the Github project, for some reason. Instead of going to the indented `haskellstack.org`, it goes to `https://github.com/NorfairKing/smos/blob/master/haskellstack.org`.

Replace with a version of the URL that includes `https://`. Also use the canonical URL, to avoid a redirect.

NorfairKing commented on smos-convert/package.yaml:
```diff
@@ -0,0 +1,71 @@
+name:                smos-convert
+version:             0.1.0.0
+github:              https://github.com/NorfairKing/smos
+license:             BSD3
```
AllRightsReserved

NorfairKing commented on smos-convert/src/Import.hs:
```diff
@@ -0,0 +1,26 @@
+
+import Prelude as X hiding (head, init, last, tail)
+
+import GHC.Generics as X hiding (to)
```
Only import `Generic` instead of only hiding `to`.

NorfairKing commented on smos-convert/src/Smos/Convert/OptParse.hs:
```diff
@@ -0,0 +1,83 @@
+    parser = ConvertFile <$> parseConvertFileArgs
+
+parseCommand :: Parser Command
+parseCommand = hsubparser $ command "convert" parseConvertFile
```
So the command will be `smos-convert convert FILE`? Why not just `smos-convert FILE`?

NorfairKing commented on smos-convert/test/Spec.hs:
```diff
@@ -0,0 +1,2 @@
+main :: IO ()
+main = putStrLn "Test suite not yet implemented"
```
Just leave it out then.

Nickske666 replied:
I don't understand why.
I added this as a reminder that I need to add tests.
What do you want to replace it by, and more importantly, what is wrong with this?

NorfairKing commented on smos-data/src/Smos/Data/Types.hs:
```diff
@@ -126,7 +126,8 @@ instance FromJSON Entry where
     toJSON Entry {..} =
-        if and [ isNothing entryContents
+        if and
+               [ isNothing entryContents
```
No cosmetic changes here please.

NorfairKing commented on smos/src/Smos/Draw.hs:
```diff
@@ -55,7 +55,8 @@ drawNoContent :: Widget n
     B.vCenterLayer $
     B.vBox $
-    map B.hCenterLayer
+    map
```
No cosmetic changes.

Nickske666 commented:
I only changed the one comment you made.
Everything is good now.

Commit:
Fix Stack link

Without the `https://` in the beginning, Github turns urls into relative links inside the Github project, for some reason. Instead of going to the indented `haskellstack.org`, it goes to `https://github.com/NorfairKing/smos/blob/master/haskellstack.org`.

Replace with a version of the URL that includes `https://`. Also use the canonical URL, to avoid a redirect.

NorfairKing commented:
Oh wauw I would never have caught that. Thank you so much!
